### ISSUE TITLE
Version 1.1 (Blender version 2.90.0) freezes when trying to import a .obj file

### ISSUE THREAD

**Issue by erwolfe** [FSWindowSeat/BlenderImportXPObj#1]
I've been trying to import a terminal building obj file from a scenery I downloaded.

The .obj references texture files in a .PNG format, but the ones the scenery came with are .dds, and when I try to import, it gives an error saying "Terminal.PNG" doesn't exist.

I thought converting the .dds files to .PNG files with Nvidia's Texture Tools Exporter would do the trick, but now when I try to import the .obj, with the .PNG files it references in the same folder, blender stops responding and I need to force quit it.

~~Unfortunately I can't find blender crash logs (possibly since it never actually crashes, I terminate the process after a while).~~
Once I waited a looong time, blender basically completely locked up my PC, but eventually crashed and output a blender.crash.txt file that I will attach.
[blender.crash.txt](https://github.com/FSWindowSeat/BlenderImportXPObj/files/5833905/blender.crash.txt)
[Pastebin if you don't want to download the txt file](https://pastebin.com/VgPdVp9R)

**Comment by FSWindowSeat** [comment]
Thanks for letting me know. Any chance you can share the .obj file so that I can test it myself, or maybe you can let me know which scenery you're referring to so that I can download and review it myself? I've never encountered this issue, so I'm curious to investigate the root cause. Thanks

**Comment by erwolfe** [comment]
I might be able to send it, how do you want me to contact you

**Comment by FSWindowSeat** [comment]
Thanks, my email address is muppetlabs[at]fswindowseat.com. I'll make sure not share it with anyone else and will make sure to delete it at the end of my investigation.

**Comment by FSWindowSeat** [comment]
I've tried to reproduce the issue at my end, but I was able to import the file successfully. However, I share your experience; Blender locked up for a good 3-5 minutes and maxed out my machine's memory (32 gb). 

I assume the file is just too large. I don't know what the author(s) did, but the .obj file included over 1 million vertices. Just for comparison, another scenery that I converted recently included roughly 50k vertices per file. So this 20x as many and will eventually lead to the performance problems you've encountered.

As this appears to be an issue with the large size of the file and not necessarily with the code, I'll close the issue. However, if anyone has any ideas on how to improve the code's performance, e.g. via multi threading, let me know.
